FrontEnd/App.js

```diff
@@ -18,21 +18,30 @@ const reducer = combineReducers({
 
 const store = createStore(reducer, install());
 
-class App extends React.Component {
-  constructor(props) {
-    super(props);
-  }
+function App(props) {
+    props.loadUserData(localStorage.getItem("userId"), localStorage.getItem("role"), localStorage.getItem("request_token"))
+    const [update, setUpdate] = useState(false);
 
-  render() {
-    const appropriateNavigation = this.props.role === "student" ? <SchuelerHauptmenue /> : <LehrerNavigation />;
-    const loginScreenOrNavigation = this.props.loggedIn ? appropriateNavigation : <LoginScreen />;
+    function handleUpdate() {
```

Comment:
Was hat das hier für einen Sinn?

Reply:
Wenn du den State ändern willst, damit sich app updatet, dann änder doch einfach was am globalen (redux) state und gib den entsprechend geänderten Teil über mapStateToProps in die Klasse App herein.

Reply:
Ich denke es müsste sogar reichen, wenn du des bei dem loginReducer mit reinmachst und da loggedIn auf false setzt, wenn die entsprechende logout action dispatched wird. Denn loggedIn wird ja schon in die props von app gemapt. Heißt wenn du das auf flase setzt müsste sich app neu rendern

Reply:
Ich hatte dieses handleUpdate tatsächlich genutzt, damit sich die App beim Login/Logout updated und die entsprechende Navigation läd. Es wurden ja vorher schon Sachen beim Login im globalstate geändert, es hätte also theoretisch automatisch schon funktionieren müssen. Hats aber nicht. deswegen hatte ichs über den umständlichsten Weg gemacht. Jetzt funktionierts auch ohne das handleUpdate. Aber kein Plan warum das vorher nicht funktionierte xD